### PR TITLE
Cleaning tests and adding workflows

### PR DESCRIPTION
**Pull request by mmatera**
Finally, all the doctests passes. The remaining tests were moved to a pytest module. Also, workflows were added.

### CODE REVIEW

**Comment by mmatera** [comment]
@rocky,
what is missing here is to adjust the workflows to make the tests. Locally, it pass all the tests or skip the failing ones.

File: pymathics/graph/base.py [hunk 1]
```diff
@@ -636,12 +638,17 @@ def empty(self):
...
 
+    def is_directed(self):
+        if self.G.is_directed():
+            return not self.mixed
+        return False
+    
     def is_loop_free(self):
         return not any(True for _ in nx.nodes_with_selfloops(self.G))
 
     # networkx graphs can't be for mixed
```
Comment (rocky) [inline pymathics/graph/base.py:649]: can't be for -> cannot be used for

**Comment by rocky** [comment]
LGTM - one very small grammar typo noticed that was there from before

File: pymathics/graph/base.py (hunk 1 above)
Comment (rocky) [reply pymathics/graph/base.py:649]: I will correct this when I rebase for https://github.com/Mathics3/pymathics-graph/pull/13

Comment (mmatera) [reply pymathics/graph/base.py:649]: Excellent. With this baseline, we can go over #13 and see what is failing.

**Pull request merged**